From: =?utf-8?q?Tim_R=C3=BChsen?= <tim.ruehsen@gmx.de>
Date: Tue, 21 Feb 2017 12:18:29 +0100
Subject: Fix order of files in psl_latest()

If 'dist_filename' and 'filename' are given and both have the same
age, we want 'dist_filename' (expected DAFSA) being loaded.

(cherry picked from commit 448f6e456489f8e3501b3c0aa79761c5a5b2b3d2)
---
 src/psl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/psl.c b/src/psl.c
index 1c8e191..4e6aa14 100644
--- a/src/psl.c
+++ b/src/psl.c
@@ -1833,8 +1833,8 @@ psl_ctx_t *psl_latest(const char *fname)
 
 	/* create array of PSL files reverse sorted by mtime (latest first) */
 	ntimes = _insert_file(fname, psl_fname, psl_mtime, 0);
-	ntimes = _insert_file(_psl_filename, psl_fname, psl_mtime, ntimes);
 	ntimes = _insert_file(_psl_dist_filename, psl_fname, psl_mtime, ntimes);
+	ntimes = _insert_file(_psl_filename, psl_fname, psl_mtime, ntimes);
 
 	/* load PSL data from the latest file, falling back to the second recent, ... */
 	for (psl = NULL, it = 0; it < ntimes; it++) {
